src/main/kotlin/org/pkl/lsp/ast/Extensions.kt

```diff
@@ -626,3 +626,102 @@ fun Appendable.renderParameterList(
 
 fun <T> List<T>.withReplaced(idx: Int, elem: T): List<T> =
   toMutableList().apply { this[idx] = elem }
+
+fun PklNode.getDocumentationUrl(): String? {
+  val module = if (this is PklModule) this else enclosingModule ?: return null
+  val file = module.containingFile
+
+  val packageDep = file.`package`
+
+  val authorityUrl =
+    when {
+      packageDep == null && file.pklAuthority == Origin.STDLIB.name.lowercase() ->
+        DEFAULT_PKL_AUTHORITY
+      packageDep != null -> packageDep.packageUri.authority
+      else -> return null
+    }
+
+  val packagePath =
+    when {
+      packageDep == null && file.pklAuthority == Origin.STDLIB.name.lowercase() -> {
+        "pkl"
+      }
+      packageDep != null -> {
+        val packageUri = packageDep.packageUri
+        val authority = packageUri.authority
+
+        authority + "/" + packageUri.path.substringBeforeLast('@').removePrefix("/")
```

Comment:
Make `org.pkl.lsp.packages.dto.PackageUri#basePath` public, and use that instead of `packageUri.path.substringBeforeLast('@')`